lib/knode.js

```diff
@@ -55,14 +55,31 @@ exports.KNode.prototype._onMessage = function(message) {
     if (!message.type || typeof message.type !== 'string')
         return;
 
-    var methodName = '_on' + _.str.titleize(_.str.camelize(message.type.toLowerCase()));
-    var action = this[methodName];
-    if (action) {
-        this._updateContact(MC(message));
-        _.bind(action, this)(message);
+    let action = null;
+    switch (message.type) {
+        case 'PING':
+            action = this._onPing();
```

Comment:
This (and others) needs to be `action = this._onPing` without the actual call, since that happens below.